tests/xharness/Jenkins/TestTasks/BuildToolTask.cs

```diff
@@ -35,8 +36,10 @@ public abstract class BuildToolTask : AppleTestTask
 			}
 		}
 
-		protected BuildToolTask (Jenkins jenkins, IProcessManager processManager) : base (jenkins)
-			=> buildToolTask = new Xharness.TestTasks.BuildToolTask (processManager);
+		protected BuildToolTask (Jenkins jenkins, IProcessManager processManager) : base (jenkins) {
+			ProcessManager = processManager ?? throw new ArgumentNullException (nameof (processManager));
+			InitializeTool ();
```

Comment:
Why having protected method like this when we can just do the initialization right in child constructors?

Reply:
because each task is creating its own tool object, and I did not like the fact that the parent class would create a tool, that later would be overriden and then will use memory until the next GC call. Felt ugly.

Reply:
Right, makes sense, I didn't realize there are some non-abstract parents in the chain

Reply:
It would also might make sense to still keep it `readonly` and only do `this.buildToolTask = InitializeTool ();`. But keep it as is, it's totally fine